### ISSUE TITLE
Add option to hide the bundle's label when you expand the bundle so other labels are more visible

### ISSUE THREAD

**Issue by jakelauer** [leggett/simplify#788]
If an email receives two labels, it will end up in two bundles at the same time. This causes some confusion for me, because my bundles have purposes - I may draw different conclusions about an email based on the bundle in which it appears. 

For instance, if I have an email which gets marked as both "Newsletter" and "Important", I might archive it without thinking if I open up my Newsletter bundle first.

A couple ideas for avoiding this:

- Allow me to configure Simplify with a prioritized label list which I fill in. 
    - If it's empty, or if I don't put both of the labels in the list, the behavior is the same as it is now. If one of the labels does appear in the list, the email goes into that bundle. If both labels appear in the list, the email goes into the highest-priority bundle
- Provide a simple toggle to choose between emails appearing in multiple bundles or just one
    - Feels like this could be problematic if an email has two labels, but only one of those has a bundle at a given time
- Show something in the UI to denote emails which are also inside another bundle
    - Seems like this would only be effective if it were accompanied by multiple ways of archiving:
        - Archive entire bundle
        - Archive this label only

Anyway, just some ideas!

---
Simplify v.3.1.3 configuration: aAX appNavOff avatarsInList avatarsZoom boldHighlight bundleInbox caA caI caL caS cfC cfO cfQ cfR cfU composeActions composeFab dateGroup defaultTheme fontList fontMsg fontNav gaArchive gaRemove groupHideFirst groupInbox groupOtherLists hideAds hideExtWarn hideFileIcons hideImportance hideInboxHideAllLink hideLabelChips hideListCount hideMsgCount hideMsgReactions hideMsgReplies hideNavLabels hideSignatures hideTabIcons importantBadge inInbox inList invertAddons invertCompose izBgDefault lightTheme listWidthXLg lowDensity maDone maLabel maMove maPin maTask matYou msgWidthSm nPane navOpen newNav newUI reminders rightImportance rightLabels rightStar sendLater simplify simplifyPrefsOpen unreadDot vPane1line- System: Mozilla/5.0 (Macintosh Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36 - gmail.pinto-server_20240104.06_p1 - Window: 1728 x 972 (100pct zoom) - Language: en - Errors: / 11589:52: Uncaught TypeError: Cannot read properties of null (reading 'previousSibling')

**Comment by leggett** [comment]
This is definitely a tough situation to handle. I don't think the first two approaches are tenable. The first one is way too complicated. The second one is too unpredictable. 

The third one could be made better if I add an option to hide the label for the expanded bundle. You would still need to expand the bundle to see what is in it but to your original scenario... You open the Newsletter bundle and instead of every message having a "Newsletter" label and one having a "Important" label (might be easy to miss), you would only see labels that aren't "Newsletter" making it more likely you see the "Important" label

There _might_ be a way to hint at this in the collapsed bundle too (maybe I show other labels to the far right) but I'm not sure. 

There are other things you might consider doing:

- Make the color of the Important label stronger so it stands out more
- Tell Simplify not to bundle anything with "Important" in which case it shouldn't be in the Newsletter bundle to begin with

**Comment by jakelauer** [comment]
I agree with everything you've said. I hoped against hope that the first one wasn't all that complex given that Simplify already provides a way to skip bundling for labels, but it sounds like they are different enough to be problematic.

Part of the reason this matters is because Gmail's filtering is honestly super rudimentary - there's no way to reorder filters without exporting them all, editing the XML file, and re-importing, and there's no way to stop processing filters. This makes it super tough to write filters to avoid multiple labels. I'm tempted to write a filter manager extension!

**Comment by leggett** [comment]
Sounds good. Do you think what I described would help your situation?

I hope to tackle filters at some point. It might even be what tips me into offering more functionality if you give me limited API access (I believe you can ask for API access to just filters).

**Comment by jakelauer** [comment]
The colors help a bit, since I know to look for it. I think that's kinda the crux of the issue - if you know to look for it, it's not so bad, but the whole point is that you might not look.

Filters *could* work well if managed appropriately. You could even implement a faux "stop processing" system for any filter thusly:

- Create a filter which applies a label (`MyFilter`) and checks the "stop processing" box
- Programmatically create a second filter with the same rules as the first, ordered immediately after it, which applies the `filter-stop` label
- Ensure all labels include a `-{label:filter-stop}` parameter
- If possible, remove the `filter-stop` label at the end.

The good news is I never gave a rat's ass about any of this until now because labels were pretty useless for me, but now that you've added Bundles, labels (and by extension, filters) are supercharged with value.

Does nesting labels affect this system at all?

**Comment by leggett** [comment]
Bundles work fine with nested labels. I don't truncate long label paths in the same way that Gmail does which makes them a little worse but I think I can fix that.

**Comment by ben-foxmoore** [comment]
> The third one could be made better if I add an option to hide the label for the expanded bundle.

@leggett did this option make it in at some point? This would be a big help IMO.

My context/use case: I have a Github label, and nested under that a few important repos, and then "mentioned", "assigned", etc. Because Gmail's built-in filters aren't ordered, it's not possible to have a "label with Github only if nothing more specific matched", so everything in the nested labels also gets labelled as "Github".

Ideally, I'd love for Simplify to detect that a sub-label is also applied, and only show it in that bundle, but I realize that's (likely much) more complicated logic to implement. This is similar to @jakelauer's second approach, which you said would be too unpredictable, but has the advantage that it's predictable based on Gmail's built-in label nesting. This would of course mean that emails would still appear in multiple bundles if they had multiple labels where one isn't strictly a parent of the other.

**Comment by leggett** [comment]
No, I never added an option to hide the label for the currently expanded bundle. I might do that but I have a number of larger things ahead of something like this that I wouldn't suggest counting on it being added any time soon. 

Gmail made some changes around how focus is managed and I've been rebuilding bundles for the last few months to handle these changes and improve performance.

I did add options to truncate label paths to just the label name (e.g., `Mentioned` instead of `GitHub/Mentioned`):
https://mail.google.com/mail/?simplify=bundles&pref=shortLabels

And I added some enhancements to Gmail's filters UI (I turned it into columns and added sorting):
<img width="1051" alt="Image" src="https://github.com/user-attachments/assets/90768f4c-5dbb-4538-a5b2-13b5f5b10022" />

Another thing I do intend to add is an option to click on a parent label in the nav and automatically do a search for not just the parent label, but all the email in any of the children labels. If you set the parent label to not bundle and turned on bundles outside the inbox, this could make the parent label in the nav into a good inbox. I have `News` as a parent label with all the newsletters I subscribe to as children labels.

**Comment by ben-foxmoore** [comment]
> I did add options to truncate label paths to just the label name (e.g., `Mentioned` instead of `GitHub/Mentioned`): [mail.google.com/mail?simplify=bundles&pref=shortLabels](https://mail.google.com/mail/?simplify=bundles&pref=shortLabels)

Yes, I use that already - thanks!

> And I added some enhancements to Gmail's filters UI (I turned it into columns and added sorting): <img alt="Image" width="1051" src="https://private-user-images.githubusercontent.com/8763662/447355391-90768f4c-5dbb-4538-a5b2-13b5f5b10022.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDgyNDcxOTEsIm5iZiI6MTc0ODI0Njg5MSwicGF0aCI6Ii84NzYzNjYyLzQ0NzM1NTM5MS05MDc2OGY0Yy01ZGJiLTQ1MzgtYTViMi0xM2I1ZjViMTAwMjIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDUyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA1MjZUMDgwODExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NmJjZmI2ZjczODk3NjM2MzA5ZWI2ZDlkZDg2M2UxM2I0MzZlMjcwN2VmOWYzMjc5NDVkNWJmNzc0NzRmMzJiMyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.itvspTsRBI6DXAA7Bsh4o9_nIZKVUPT3TEojgs5zRJs">

I didn't even realize that was a Simplify addition 😁 


> Another thing I do intend to add is an option to click on a parent label in the nav and automatically do a search for not just the parent label, but all the email in any of the children labels. If you set the parent label to not bundle and turned on bundles outside the inbox, this could make the parent label in the nav into a good inbox. I have `News` as a parent label with all the newsletters I subscribe to as children labels.

Oh this sounds useful!
